airbyte-integrations/connectors/source-klarna/source_klarna/source.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import urlparse, parse_qs
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import BasicHttpAuthenticator
+
+
+# Basic full refresh stream
+class KlarnaStream(HttpStream, ABC):
+    def __init__(self, region: str, playground: bool, authenticator: BasicHttpAuthenticator, **kwargs):
+        self.region = region
+        self.playground = playground
+        self.kwargs = kwargs
+        super().__init__(authenticator=authenticator)
+
+    page_size = 500
+
+    @property
+    def url_base(self) -> str:
+        playground_path = 'playground.' if self.playground else ''
+        if self.region == 'eu':
+            endpoint = f"https://api.{playground_path}klarna.com/"
+        else:
+            endpoint = f"https://api-{self.region}.{playground_path}klarna.com/"
+        return endpoint
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_json = response.json()
+        if 'pagination' in response_json:
+            if 'next' in response_json['pagination']:
+                parsed_url = urlparse(response_json['pagination']['next'])
+                query_params = parse_qs(parsed_url.query)
+                # noinspection PyTypeChecker
+                return query_params
+        else:
+            return None
```

Comment:
```suggestion
        if 'next' in response_json.get('pagination', {}).keys():
                parsed_url = urlparse(response_json['pagination']['next'])
                query_params = parse_qs(parsed_url.query)
                # noinspection PyTypeChecker
                return query_params
        else:
            return None
```

You can simplify the code.

Reply:
Also add a unit test to validate the function.